core/src/main/java/hudson/slaves/WorkspaceList.java

```diff
@@ -302,6 +303,7 @@ public void release() {
      * @return a sibling directory, for example {@code …/something@tmp} for {@code …/something}
      * @since 1.652
      */
+    @CheckForNull
```

Comment:
Sort of incompatible for plugin callers, though in cases where this now returns null, it would previously have thrown a `NullPointerException`, so maybe this is no worse.

Reply:
Some Javadoc could be moved over from https://github.com/jenkinsci/jenkins/pull/4526

Reply:
Kept it brief, but yeah.